.circleci/config.yml

```diff
@@ -0,0 +1,182 @@
+
+  pytest:
+    description: |
+      Run pytest with code coverage.
+      Store the coverage reports as artifacts.
+      Fail if the code coverage is below the minimum threshold.
+    parameters:
+      coverage_path:
+        description: Filesystem path to measure coverage.
+        type: string
+        default: .
+
+      min_coverage:
+        description: Minimum code coverage percentage ([0,100]).
+        type: integer
+        default: 100
+
+    steps:
+      - run:
+          name: "[python] Run pytest with code coverage"
+          command: |
+            pytest \
+              --capture=no \
+              --cov-branch \
+              --cov-fail-under=<<parameters.min_coverage>> \
+              --cov-report=html:coverage-reports \
+              --cov=<<parameters.coverage_path>> \
+              --junitxml=coverage-reports/junit.xml \
+              --ignore=common \
+      - store_test_results:
+          path: test-reports
+      - store_artifacts:
+          path: test-reports
+      - store_artifacts:
+          path: coverage-reports
+
+jobs:
+  test:
+    docker:
+      - image: cimg/python:3.8.2
+      - image: circleci/mongo:4.0.18-xenial
+      - image: circleci/redis:5.0.9-alpine3.11
+    resource_class: small
+    steps:
+      - checkout
+      - initialize-submodules
+      - poetry-install:
+          deps_cache_version: "v2"
+      - pytest:
+          coverage_path: ./immuni_app_configuration
+
+  check:
+    docker:
+      - image: cimg/python:3.8.2
+    resource_class: small
+    steps:
+      - checkout
+      - initialize-submodules
+      - poetry-install:
+          deps_cache_version: "v2"
+      - run:
+          name: "[python] Run checks"
+          command: poetry run checks immuni_app_configuration --ci
+
+  build-and-run-example:
+    docker:
+      - image: cimg/base:2020.05
+    resource_class: small
+    steps:
+      - checkout
+      - initialize-submodules
+      - setup_remote_docker
+      - run:
+          name: "[docker] Build the images"
+          command: |
+            cd docker && docker-compose build \
+                --build-arg GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD) \
+                --build-arg GIT_SHA=$(git rev-parse --verify HEAD) \
+                --build-arg GIT_TAG=$(git tag --points-at HEAD | cat) \
+                --build-arg BUILD_DATE=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+      - run:
+          name: "[docker] Run the example"
+          command: cd docker && docker-compose up --detach
+      - run:
+          name: "[docker] Health check"
+          command: |
+            docker run \
+                --network container:docker_api_1 \
+                appropriate/curl \
+                --verbose \
+                --retry 10 \
+                --retry-delay 1 \
+                --retry-connrefused \
+                http://localhost:5000
+      - run:
+          name: "[docker] Stop the example"
+          command: cd docker && docker-compose down
```

Comment:
do we actually need this step to stop the daemon? 🤔

Reply:
Not really, it was mostly to be explicit. Anyway, we can let CircleCI deal with getting rid of it. Let me remove it, thanks! 🚀